test/libgit2.jl

```diff
@@ -1013,18 +1013,22 @@ mktempdir() do dir
             end
 
             # Find a hostname that maps to the loopback address
-            hostname = replace(readchomp(`hostname`), r"\..*$", "")
-            loopback = ip"127.0.0.1"
+            hostnames = ["localhost"]
+            try
+                # In minimal environments a hostname might not be available (issue #20758)
+                unshift!(hostnames, replace(gethostname(), r"\..*$", ""))
```

Comment:
why is it this is stripping anything after the last dot? (if I'm reading that regex right)